teku/src/main/java/tech/pegasys/teku/cli/subcommand/RemoteSpecLoader.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.cli.subcommand;
+
+import java.net.URI;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import tech.pegasys.teku.infrastructure.exceptions.InvalidConfigurationException;
+import tech.pegasys.teku.infrastructure.logging.SubCommandLogger;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.SpecFactory;
+import tech.pegasys.teku.spec.config.SpecConfigLoader;
+import tech.pegasys.teku.validator.remote.apiclient.OkHttpClientAuthLoggingIntercepter;
+import tech.pegasys.teku.validator.remote.apiclient.OkHttpValidatorRestApiClient;
+
+class RemoteSpecLoader {
+
+  private static final long RETRY_DELAY = 5000;
+
+  static Spec getSpec(OkHttpValidatorRestApiClient apiClient) {
+    try {
+      return apiClient
+          .getConfigSpec()
+          .map(response -> SpecConfigLoader.loadConfig(response.data))
+          .map(SpecFactory::create)
+          .orElseThrow();
+    } catch (Exception e) {
+      String errMsg =
+          String.format(
+              "Failed to retrieve network spec from beacon node endpoint '%s'.\nDetails: %s",
+              apiClient.getBaseEndpoint(), e.getMessage());
+      throw new InvalidConfigurationException(errMsg, e);
+    }
+  }
+
+  static Spec getSpec(URI beaconEndpoint) {
+    return getSpec(createApiClient(beaconEndpoint));
+  }
+
+  static Spec getSpecWithRetry(URI beaconEndpoint) {
+    return retry(() -> getSpec(beaconEndpoint));
+  }
+
+  static Spec getSpecWithRetry(OkHttpValidatorRestApiClient apiClient) {
+    return retry(() -> getSpec(apiClient));
+  }
+
+  private static <T> T retry(Callable<T> f) {
+    try {
+      return f.call();
+    } catch (Throwable e) {
+      logError(e);
+      sleep();
+      return retry(f);
+    }
+  }
+
+  private static void sleep() {
+    try {
+      Thread.sleep(RETRY_DELAY);
+    } catch (InterruptedException e) {
+      logError(e);
+      throw new RuntimeException(e);
+    }
+  }
+
+  private static void logError(Throwable e) {
+    SubCommandLogger.SUB_COMMAND_LOG.error(e.getMessage());
+  }
```

Comment:
generally speaking the sub command logger is only being used for those commands that are standalone and not able to take advantage of the full logging capabilities...
I think here we should be able to use the full logging capabilities, given it's inside the validator-client.
To do that, we'd be doing:
```
private static final Logger LOG = LogManager.getLogger();
```
in the class, and then just using `LOG.error` etc in code, so for example
```
LOG.error("sleep interrupted while attempting to get remote spec", e);
```

I'm not sure which scenarios this would happen - potentially it's really only on shutdown, in which case i'd just make this be a LOG.debug, and not throw the runtimeException... This would avoid having big exceptions printed during an exit which is potentially just because a user decided to 'ctrl+c' the process...

Reply:
Just a follow up after a chat with Adrian... I've been filled in on the use of the sub_command_logger here. I'll add a comment on the actual sleep function which will hopefully be clearer than continuing this thread, and get us all closer to something that is going to tick all the boxes... 
For now disregard this thread given your previous discussions... sorry for the confusion!